manager/controllers/app/plotter_controller.go

```diff
@@ -120,17 +124,241 @@ func (r *PlotterReconciler) reconcileFinalizers(plotter *app.Plotter) error {
 	return nil
 }
 
+// PlotterModulesSpec consists of module details extracted from the Plotter structure
+type PlotterModulesSpec struct {
+	ClusterName     string
+	VaultAuthPath   string
+	AssetID         string
+	ModuleName      string
+	ModuleArguments *app.StepParameters
+	FlowType        app.DataFlow
+	Chart           app.ChartSpec
+	Scope           app.CapabilityScope
+}
+
+// convertPlotterModuleToBlueprintModule converts an object of type PlotterModulesSpec to type modules.ModuleInstanceSpec
+func (r *PlotterReconciler) convertPlotterModuleToBlueprintModule(plotter *app.Plotter, plotterModule PlotterModulesSpec) *modules.ModuleInstanceSpec {
+	assetIDs := []string{plotterModule.AssetID}
+	blueprintModule := &modules.ModuleInstanceSpec{
+		Chart:    &plotterModule.Chart,
+		AssetIDs: assetIDs,
+		Args: &app.ModuleArguments{
+			Copy:  nil,
+			Read:  nil,
+			Write: nil,
+		},
+		ClusterName: plotterModule.ClusterName,
+		ModuleName:  plotterModule.ModuleName,
+		Scope:       plotterModule.Scope,
+	}
+
+	if plotterModule.ModuleArguments == nil {
+		return blueprintModule
+	}
+
+	switch plotterModule.FlowType {
+	case app.ReadFlow:
+		var dataStore *app.DataStore
+		if plotterModule.ModuleArguments.Source.AssetID != "" {
+			assetID := plotterModule.ModuleArguments.Source.AssetID
+			// Get source from plotter assetID list
+			assetInfo := plotter.Spec.Assets[assetID]
+			dataStore = &assetInfo.DataStore
+			// Update vaultAuthPath from the cluster metadata
+			vaultCreds := dataStore.Vault["read"]
+			vaultCreds.AuthPath = plotterModule.VaultAuthPath
+			dataStore.Vault["read"] = vaultCreds
+		} else {
+			// Fill in the DataSource from the step arguments
+			dataStore = &app.DataStore{
+				Connection: *serde.NewArbitrary(plotterModule.ModuleArguments.Source.API),
+				Format:     plotterModule.ModuleArguments.Source.API.Format,
+			}
+		}
+		blueprintModule.Args.Read = []app.ReadModuleArgs{
+			{
+				Source:          *dataStore,
+				AssetID:         plotterModule.AssetID,
+				Transformations: plotterModule.ModuleArguments.Actions,
+			},
+		}
+	case app.WriteFlow:
+		blueprintModule.Args.Write = []app.WriteModuleArgs{
+			{
+				Destination:     plotter.Spec.Assets[plotterModule.ModuleArguments.Sink.AssetID].DataStore,
+				AssetID:         plotterModule.AssetID,
+				Transformations: plotterModule.ModuleArguments.Actions,
+			},
+		}
+	case app.CopyFlow:
+		var dataStore *app.DataStore
+		if plotterModule.ModuleArguments.Source.AssetID != "" {
+			assetID := plotterModule.ModuleArguments.Source.AssetID
+			// Get source from plotter assetID list
+			assetInfo := plotter.Spec.Assets[assetID]
+
+			dataStore = &assetInfo.DataStore
+			// Update vaultAuthPath from the cluster metadata
+			vaultCreds := dataStore.Vault["copy"]
```

Comment:
Why do we store the credentials for a copy as copy? It should be read for source and write for destination. No need to introduce yet another credential type.

Reply:
@simanadler please see this comment by @froesef which motivated him to add ReadCredentialKey and WriteCredentialKey.

Reply:
@froesef What is the reasoning for this?

Reply:
@revit13 I agree with @froesef comment. There shouldn't be copy credentials.  There should be read, write, and in the future delete.  When reading the data the read credentials should be used, and when writing the write credentials are used.

Reply:
ReadCredentialKey and WriteCredentialKey are removed.